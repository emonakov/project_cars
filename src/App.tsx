import React, { lazy, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import ContextProvider from './components/Context';
import Header from './components/Header';
import Footer from './components/Footer';
import Fallback from './components/Fallback';
import theme from './config/theme';

const Home = lazy(() => import('./components/Page/Home'));
const CarDetails = lazy(() => import('./components/Page/CarDetails'));

const App: React.FC = () => (
    <ThemeProvider theme={theme}>
        <Router>
            <Suspense fallback={<Fallback />}>
                <Header />
                <ContextProvider>
                    <Switch>
                        <Route path="/" exact component={Home} />
                        <Route path="/car/:stockId(\d+)" exact component={CarDetails} />
                    </Switch>
                </ContextProvider>
                <Footer />
            </Suspense>
        </Router>
    </ThemeProvider>
);

export default App;
