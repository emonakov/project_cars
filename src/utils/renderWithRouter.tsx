import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';

const history = createMemoryHistory();

const renderWithRouter = (Component: JSX.Element, initialRoute = '/') => {
    history.push(initialRoute);

    return render(
        <Router history={history}>
            {Component}
        </Router>,
    );
};

export default renderWithRouter;
