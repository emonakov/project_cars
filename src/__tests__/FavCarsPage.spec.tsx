import React from 'react';
import { screen } from '@testing-library/react';

import App from '../App';
import renderWithRouter from '../utils/renderWithRouter';
import cars from './__mock__/cars.mock.json';

describe('Tests favorites page', () => {
    afterEach(() => {
        window.localStorage.clear();
    });

    it('Renders the favorites page without cars in the local storage', async () => {
        renderWithRouter(<App />, '/car/fav');

        const carsCounter = await screen.findByText(/0 cars in this list/i);
        expect(carsCounter).toMatchSnapshot();
    });

    it('Renders the favorites page with cars in the local storage', async () => {
        window.localStorage.setItem('favCars', JSON.stringify([...cars.BMW.cars]));
        renderWithRouter(<App />, '/car/fav');
        const carsCounter = await screen.findByText(`${cars.BMW.cars.length} cars in this list`);

        expect(carsCounter).toMatchSnapshot();
    });
});
