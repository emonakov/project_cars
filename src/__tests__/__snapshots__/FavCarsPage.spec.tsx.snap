// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests favorites page Renders the favorites page with cars in the local storage 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
>
  10 cars in this list
</p>
`;

exports[`Tests favorites page Renders the favorites page without cars in the local storage 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
>
  0 cars in this list
</p>
`;
